.github/workflows/build.yml

```diff
@@ -51,15 +51,28 @@ jobs:
           name: dist
           path: dist
 
+  build-test-matrix:
+    needs: build
+    runs-on: ubuntu-latest
+    outputs:
+      tests: ${{ steps.set-tests.outputs.tests }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - id: set-tests
+        run: |
+          tests=$(cd test && npx -p jest -p ts-jest jest --listTests | jq  -R -s -c 'split("\n") | map(select(length > 0)) | { target: values, terraform: ["0.14.11", "0.15.4"]}')
+          echo $tests
+          echo "::set-output name=tests::$tests"
+
   linux_integration:
+    needs: build-test-matrix
     runs-on: ubuntu-latest
     strategy:
-      matrix:
-        terraform: ["0.14.11", "0.15.4"]
-        target: ["typescript", "python", "java", "csharp", "go"]
+      fail-fast: false
+      matrix: ${{fromJSON(needs.build-test-matrix.outputs.examples)}}
```

Comment:
Should this be ...outputs.tests instead of ...outputs.examples?

Reply:
Yes, thanks!